Avoid an implicit declaration of dup2 in its configure probe.  This
prevents build issues with future compilers.

Already reported upstream:

  <https://lists.gnu.org/archive/html/bug-bash/2023-02/msg00000.html>

diff --git a/aclocal.m4 b/aclocal.m4
index 6162f6eb9ef90754..5cbbe96d7158197a 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -238,6 +238,9 @@ AC_CACHE_VAL(bash_cv_dup2_broken,
 #include <sys/types.h>
 #include <fcntl.h>
 #include <stdlib.h>
+#ifdef HAVE_UNISTD_H
+# include <unistd.h>
+#endif /* HAVE_UNISTD_H */
 int
 main()
 {
